### PR TITLE
Basic WiFi Configuration and API 

### PR DESCRIPTION
**Pull request by ghost**
Adds basic WiFi configuration functionality and an API to toggle previously specified pins `on` or `off` and to get the current status (`on`/`off`/`undefined`) of the specified pins.

For details see the readme. I will add the most imporant info to the main readme later.

### CODE REVIEW

File: firmware/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Juicy Switch Firmware 🔌 🍊 💻
+
+⚠️ **This is work in progress!** ⚠️
```
Comment (Rogger) [inline firmware/README.md:3]: Neat!

File: firmware/platformio.ini [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+;
+;   Build options: build flags, source filter
+;   Upload options: custom upload port, speed and extra flags
+;   Library options: dependencies, extra library storages
+;   Advanced options: extra scripting
+;
+; Please visit documentation for the other options and examples
+; http://docs.platformio.org/page/projectconf.html
+
+[env:d1_mini]
```
Comment (Rogger) [inline firmware/platformio.ini]: I had to specify the following in order to get it working with ESP-07
```
[env:esp07]
platform = espressif8266
board = esp07
framework = arduino
```

File: firmware/src/API/API.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+API::~API() {
+	server.reset();
+}
+
+void API::addPin(uint8_t pin, uint8_t valueOn, uint8_t valueOff) {
+	pins.push_back(std::make_tuple(pin, valueOn, valueOff));
+}
+
+void API::setup() {
```
Comment (Rogger) [inline firmware/src/API/API.cpp:17]: 👍

File: firmware/src/Config/HttpHandler.cpp [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    	int c = str.charAt(i);
+    	if (c != '.' && (c < '0' || c > '9')) {
+      		return false;
+    	}
+  	}
+
+  	return true;
+}
+
+String HttpHandler::toString(IPAddress ip) {
```
Comment (Rogger) [inline firmware/src/Config/HttpHandler.cpp:126]: 🤓

File: firmware/README.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+The firmware powering Juicy's *esp* chip. It lets you configure the WiFi that Juicy should connect to and provides a RESTful API for controlling the chip's LED. Later, when Juicy is fully assembled, you will be able to power the socket on and off via that API.
```
Comment (Rogger) [inline firmware/README.md]: Should we call it socket or switch? What do you prefer?

**Review by Rogger — APPROVED** [review]
Super clean structure and very well done, nice 👐

File: firmware/README.md (hunk 5 above)
Comment (ghost) [reply firmware/README.md]: Let's do *switch*. Sounds easier.

**Pull request merged**